broker/neb/src/set_log_data.cc

```diff
@@ -224,3 +275,628 @@ void neb::set_log_data(neb::log_entry& le, char const* log_data) {
   le.host_id = engine::get_host_id(le.host_name);
   le.service_id = engine::get_service_id(le.host_name, le.service_description);
 }
+
+/**
+ *  Extract Nagios-formated log data to the C++ object.
+ *
+ *  Return true on success.
+ */
+bool neb::set_pb_log_data(neb::pb_log_entry& le, const std::string& output) {
+  // Duplicate string so that we can split it with strtok_r.
+  auto& le_obj = le.mut_obj();
+
+  // First part is the log description.
+  auto s = absl::StrSplit(output, absl::MaxSplits(':', 1));
+  auto it = s.begin();
+  auto typ = *it;
+  ++it;
+  auto lasts = *it;
+  lasts = StripLeadingAsciiWhitespace(lasts);
+  auto args = absl::StrSplit(lasts, ';');
+  auto ait = args.begin();
+
+  if (typ == "SERVICE ALERT") {
+    le_obj.set_msg_type(LogEntry_MsgType_SERVICE_ALERT);
+    if (ait == args.end()) {
+      log_v2::neb()->error("Missing host name in log message '{}'",
+                           le_obj.output());
+      return false;
+    }
+    le_obj.set_host_name(std::string(ait->data(), ait->size()));
```

Comment:
you can avoid std::string with le_obj.set_host_name(ait->data(), ait->size());

Reply:
:+1: